.github/workflows/benchmark_long.yml

```diff
@@ -0,0 +1,499 @@
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          # We need the parent commit to do a diff
+          fetch-depth: 2
+      - name: "Fetching PR description"
+        # We fetch the latest pull request data (description, labels, ...) from
+        # API instead of using stale one from pull_request event. This makes it
+        # possible to update the trailers, labels on the pull request and re-run
+        # the workflow to make them take effect.
+        # This is majorly for triggering benchmarks without pushing new commits.
+        # See https://github.com/openxla/iree/issues/10042#issuecomment-1449250094
+        # for more details.
+        id: fetch-pr
+        if: github.event_name == 'pull_request'
+        env:
+          PR_NUMBER: ${{ github.event.number }}
+          PR_JSON: pull_request.json
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh api "/repos/${GITHUB_REPOSITORY}/pulls/${PR_NUMBER}" > "${PR_JSON}"
+          # It requires delimiter to pass multiline strings through
+          # GITHUB_OUTPUT. Since these are already escaped JSON strings, pass
+          # the JSON strings and later use fromJSON to decode them.
+          echo "pr-title=$(jq '.title' ${PR_JSON})" >> "${GITHUB_OUTPUT}"
+          echo "pr-body=$(jq '.body' ${PR_JSON})" >> "${GITHUB_OUTPUT}"
+          # Use --compact-output to avoid multiline JSON.
+          echo "pr-labels=$(jq --compact-output '.labels | map(.name)' \
+            ${PR_JSON})" >> "${GITHUB_OUTPUT}"
+      - name: "Configuring CI options"
+        id: configure
+        env:
+          PR_TITLE: ${{ fromJSON(steps.fetch-pr.outputs.pr-title || '""') }}
+          PR_BODY: ${{ fromJSON(steps.fetch-pr.outputs.pr-body || '""') }}
+          PR_LABELS: ${{ steps.fetch-pr.outputs.pr-labels || '[]' }}
+          ORIGINAL_PR_TITLE: ${{ github.event.pull_request.title }}
+          ORIGINAL_PR_BODY: ${{ github.event.pull_request.body }}
+          ORIGINAL_PR_LABELS: ${{ toJSON(github.event.pull_request.labels.*.name) }}
+        run: |
+          # Just informative logging. There should only be two commits in the
+          # history here, but limiting the depth helps when copying from a local
+          # repo instead of using checkout, e.g. with
+          # https://github.com/nektos/act where there will be more.
+          git log --oneline --graph --max-count=3
+
+          ./build_tools/github_actions/configure_ci.py
+
+          # Override benchmark preset.
+          echo "benchmark-presets=cuda-long" >> $GITHUB_OUTPUT
+
+      - name: "Show benchmark presets"
+        env:
+          BENCHMARK_PRESETS: ${{ steps.configure.outputs.benchmark-presets }}
+        run: |
+          echo ":stopwatch: Enabled benchmarks: \`${BENCHMARK_PRESETS}\`" \
+            >> "${GITHUB_STEP_SUMMARY}"
+
+  build_all:
```

Comment:
Ah, sad that we have to do this again :-( We should consider indexing things based on commit instead of run ID maybe, although there were reasons I didn't do that. I guess ccache will help.

It would be nice to turn this into a reusable workflow though. There's a fair amount of detail. Could you try and see how that works? My recollection is that passing inputs and outputs around was annoying though

Reply:
This is a big refactor. I'll see what I can do.

Reply:
Refactored the common parts to separate workflow files.